pkg/controller/sparkapplication/controller.go

```diff
@@ -384,27 +389,30 @@ func (c *Controller) getAndUpdateExecutorState(app *v1beta2.SparkApplication) er
 	var executorApplicationID string
 	for _, pod := range pods {
 		if util.IsExecutorPod(pod) {
```

Comment:
I actually tried this apporach and the `||` and assign variable did not play well so I just added a private method on the controller

```golang
// exceedsProcessingLimit checks whether the executor number is higher than the `executorsProcessingLimit` we want to stop persisting executors
func (c *Controller) exceedsProcessingLimit(pod *apiv1.Pod) bool {
	executorID, _ := strconv.Atoi(getSparkExecutorID(pod));
	return executorID <= c.executorsProcessingLimit
}
```

The guard clause then becomes
```golang
		if util.IsDriverPod(pod) || c.exceedsProcessingLimit(pod) {
			continue
		}
```

Reply:
looks much better, I'll refactor it. thanks!

Reply:
I also just noticed something as well.

We don't really need the `util.IsDriverPod(pod)` check  since the list of pods is always only going to be executor pods.

```golang
	pods, err := c.getExecutorPods(app)
```
And looking at the query for executor pods they are filtered at the api level.
`matchLabels[config.SparkRoleLabel] = config.SparkExecutorRole`